src/components/PopupWithForm.js

```diff
@@ -1,32 +1,40 @@
 import Popup from "./Popup.js";
-import FormValidator from "./FormValidator.js";
 
 export default class PopupWithForm extends Popup {
   constructor({ popupSelector, handleFormSubmit }) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._popupInputs = this._popupForm.querySelectorAll(".modal__input");
+    this._popupButton = this._popupForm.querySelector(".modal__button");
     this._handleFormSubmit = handleFormSubmit;
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._popupButton.textContent = "Saving...";
+    } else {
+      this._popupButton.textContent = "Submit";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```